eric_redis_queues/__init__.py

```
@@ -53,27 +54,37 @@ def setup_by_dict(self, setup: dict):
+
+        if not self._client.exists(f'{_PREFIX_QUEUES}:{self.kv_key}'):
             raise NoMessagesException
 
         try:
-            raw_value = self.__client.lpop(f'{_PREFIX_QUEUES}:{self.kv_key}')
-            return loads(raw_value)
+            raw_value = self._client.lpop(f'{_PREFIX_QUEUES}:{self.kv_key}')
+            return loads(bytes(raw_value))
 
         except Exception as e:
             raise RepositoryError(e)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)